tests/Form/UserFormsRequiredFieldsTest.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+namespace SilverStripe\UserForms\Tests\Form;
+
+use SilverStripe\CMS\Controllers\ModelAsController;
+use SilverStripe\Dev\Debug;
+use SilverStripe\Dev\SapphireTest;
+use SilverStripe\UserForms\Form\UserFormsRequiredFields;
+use SilverStripe\UserForms\Model\UserDefinedForm;
+use SilverStripe\UserForms\Form\UserForm;
+
+class UserFormsRequiredFieldsTest extends SapphireTest
```

Comment:
Didn't look at the details, but the test coverage looks pretty thorough at first glance.